lib/service/cfg.go

```diff
@@ -477,6 +481,40 @@ func (c ProxyConfig) KubeAddr() (string, error) {
 	return u.String(), nil
 }
 
+// publicPeerAddr attempts to returns the public address the proxy advertises
+// for proxy peering clients if available. It falls back to PeerAddr othewise.
+func (c ProxyConfig) publicPeerAddr() (utils.NetAddr, error) {
```

Comment:
My 2c: I would keep the pointer return (`*utils.NetAddr`), I find it less awkward to deal with when we are returning an error. (Less need to dereference variables in the body too.)

Same for `peerAddr` below.